### PR TITLE
agent: Fix is_signal_handled failing parsing str to u64

### PR DESCRIPTION
**Pull request by Champ-Goblem**
In the is_signal_handled function, when parsing the hex string returned
from `/proc/<pid>/status` the space/tab character after the colon
is not removed.

This patch trims the result of SigCgt so that
all whitespace characters are removed. It also extends the existing
test cases to check for this scenario.

Fixes: #4250
Signed-off-by: Champ-Goblem <cameron@northflank.com>

### CODE REVIEW

**Comment by egernst** [comment]
Good find @Champ-Goblem -- is it feasible to add some unit tests to exercise this?

File: src/agent/src/rpc.rs [hunk 1]
```diff
@@ -1682,7 +1682,10 @@ fn is_signal_handled(proc_status_file: &str, signum: u32) -> bool {
                 warn!(sl!(), "parse the SigCgt field failed");
                 return false;
             }
-            let sig_cgt_str = mask_vec[1];
+            let mut sig_cgt_str = mask_vec[1];
```
Comment (egernst) [inline src/agent/src/rpc.rs:1685]: I wonder if we should introduce a function here for parsing the field (maybe starting with [1679](https://github.com/kata-containers/kata-containers/pull/4252/files#diff-f87c1c212bba3abd0b07bafe69d5b01ad00dfe2c01bed585e2a75fcbb6442b89L1679)?), which would allow us to be able to add same unit tests?

File: src/agent/src/rpc.rs [hunk 2]
```diff
@@ -1682,7 +1682,10 @@ fn is_signal_handled(proc_status_file: &str, signum: u32) -> bool {
...
+            if !sig_cgt_str.chars().nth(0).unwrap().is_numeric() {
```
Comment (fidencio) [inline src/agent/src/rpc.rs:1686]: Out of curiosity, can't use simply  trim()` the string?

If that works, that would be cleaner, IMHO.

Note: https://doc.rust-lang.org/std/primitive.str.html#method.trim

File: src/agent/src/rpc.rs [hunk 3]
```diff
@@ -1682,7 +1682,10 @@ fn is_signal_handled(proc_status_file: &str, signum: u32) -> bool {
...
+                sig_cgt_str = &sig_cgt_str[1..sig_cgt_str.len()-1]
```
Comment (cmaf) [inline src/agent/src/rpc.rs:1687]: @Champ-Goblem if you end up sticking with this approach could you please update the format? Our static check failed on `cargo fmt -- --check`
```
-                sig_cgt_str = &sig_cgt_str[1..sig_cgt_str.len()-1]
+                sig_cgt_str = &sig_cgt_str[1..sig_cgt_str.len() - 1]
```

File: src/agent/src/rpc.rs (hunk 2 above)
Comment (Champ-Goblem) [reply src/agent/src/rpc.rs]: Nice spot, trim works perfectly, I have updated the code to use this

File: src/agent/src/rpc.rs (hunk 3 above)
Comment (Champ-Goblem) [reply src/agent/src/rpc.rs]: I ended up using .trim in the end but I ran this against the new commit and there should be no issues now with formatting

**Comment by Champ-Goblem** [comment]
@egernst I see there already exists some tests for this function `test_is_signal_handled`, it doesnt look like it has any cases to cover a whitespace so I have extended this to include some

File: src/agent/src/rpc.rs (hunk 1 above)
Comment (Champ-Goblem) [reply src/agent/src/rpc.rs]: This should be covered by `test_is_signal_handled` already, let me know if you would prefer to split the function down and change the related tests

**Review by fidencio — APPROVED** [review]
lgtm, thanks @Champ-Goblem!

**Review by cmaf — APPROVED** [review]

**Comment by fidencio** [comment]
Hmm.  CIs are failing to find the C version of virtiofsd and I **think** it would be easily solved with a rebase.

**Comment by fidencio** [comment]
Okay, I was able to rebase it myself.
Let me try again ...

/test

**Comment by fidencio** [comment]
```
time="2022-05-17T07:47:28-04:00" level=error msg="unable to delete test" error="write vsock host(2):1562314465->vm(256215773):1024: cannot allocate memory: unknown"
ctr: write vsock host(2):1562314465->vm(256215773):1024: cannot allocate memory: unknown
Ending hypervisor stability test
[hypervisor_stability_kill_test.sh:233] INFO: Wait until the containers gets removed
ctr: write vsock host(2):1562314465->vm(256215773):1024: cannot allocate memory: unknown
make: *** [Makefile:73: stability] Error 1
```

Power CI failure is unrelated.
